sds_data_manager/lambda_code/SDSCode/indexer.py

```diff
@@ -215,16 +245,94 @@ def batch_event_handler(event):
     dict
         HTTP response
     """
-
-    if event["detail-type"] != "Batch Job State Change":
-        return http_response(status_code=400, body="Invalid event")
+    # splited command looks like this:
+    # [
+    #   '',
+    #   'instrument codice ',
+    #   'level l1a ',
+    #   "s3_uri 's3://data-bucket/path/filename'",
+    #   "dependency [{'instrument': 'hit', 'level': 'l0', 'version': 'v00-01'}]"
+    #  ]
+    command = event["detail"]["container"]["command"][0].split("--")
+
+    # Get event inputs ready
+    s3_uri = command[3].replace("s3_uri '", "").replace("'", "")
+    filename = os.path.basename(s3_uri)
+
+    # TODO: post demo, revisit this and improve it
     if event["detail"]["status"] == "SUCCEEDED":
-        # TODO:
-        # Get filename and then
-        # write it to file catalog table
-        pass
+        # Frist write to status table and then
+        # write to file catalog with foreign key
+        # information
+        try:
```

Comment:
A couple of comments here.

* Should we move the try/except up to the very highest levels of these handlers and surround the entry-point into the function? Then we wouldn't need to try/except all the various blocks.
```python
def handler(event, context):
    try:
        return sub_handler(event, context)
    except Except as e:
        return http_response(status_code=400, body=str(e))
```

* You might want to move the if/elif inside the Session() calls, so you don't need to repeat quite as much boilerplate.

Reply:
I will visit that after demo too.